android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/dao/FamilyRegisterDao.kt

```diff
@@ -0,0 +1,243 @@
+      val members =
+        family.member?.map { member ->
+          fhirEngine.load(Patient::class.java, member.entity.extractId()).let { patient ->
+            loadFamilyMemberRegisterData(patient.logicalId)
+          }
+        }
+          ?: listOf()
+      val familyHead = loadFamilyHead(family)
+
+      RegisterData.FamilyRegisterData(
+        id = family.logicalId,
+        name = family.name,
+        identifier =
+          family.identifier
+            .firstOrNull {
+              it.use.name.contentEquals(DefaultPatientRegisterDao.OFFICIAL_IDENTIFIER)
+            }
+            ?.value,
+        address = familyHead?.extractAddress() ?: "",
+        head = familyHead?.let { loadFamilyMemberRegisterData(familyHead.logicalId) },
+        members = members,
+        servicesDue = members.sumOf { it.servicesDue ?: 0 },
+        servicesOverdue = members.sumOf { it.servicesOverdue ?: 0 },
+        lastSeen = family.meta?.lastUpdated.lastSeenFormat()
+      )
+    }
+  }
+
+  override suspend fun loadProfileData(appFeatureName: String?, resourceId: String): ProfileData {
+    val family = fhirEngine.load(Group::class.java, resourceId)
+    val familyHead = loadFamilyHead(family)
+
+    return ProfileData.FamilyProfileData(
+      id = family.logicalId,
+      name = family.name,
+      identifier =
+        family.identifier
+          .firstOrNull { it.use.name.contentEquals(DefaultPatientRegisterDao.OFFICIAL_IDENTIFIER) }
+          ?.value,
+      address = familyHead?.extractAddress() ?: "",
+      head = familyHead?.let { loadFamilyMemberProfileData(familyHead.logicalId) },
+      members =
+        family.member?.map { member ->
+          fhirEngine.load(Patient::class.java, member.entity.extractId()).let { patient ->
+            loadFamilyMemberProfileData(patient.logicalId)
+          }
+        }
+          ?: listOf(),
+      services =
+        defaultRepository.searchResourceFor<CarePlan>(
+          subjectId = family.logicalId,
+          subjectParam = CarePlan.SUBJECT,
+          filters = getRegisterDataFilters(FAMILY_CARE_PLAN)
+        ),
+      tasks =
+        fhirEngine.search<Task> {
+          filterByResourceTypeId(Task.SUBJECT, ResourceType.Group, family.logicalId)
+        }
+    )
+  }
+
+  override suspend fun countRegisterData(appFeatureName: String?): Long {
+    // TODO fix this workaround for groups count
+    return fhirEngine
+      .search<Group> { getRegisterDataFilters(FAMILY.name).forEach { filterBy(it) } }
+      .filter { it.active && !it.name.isNullOrEmpty() }
+      .size
+      .toLong()
+  }
+
+  private suspend fun loadFamilyHead(family: Group) =
+    family.managingEntity?.let { reference ->
+      fhirEngine
+        .search<RelatedPerson> {
+          filterByResourceTypeId(
+            RelatedPerson.RES_ID,
+            ResourceType.RelatedPerson,
+            reference.extractId()
+          )
+        }
```

Comment:
This is fine, but it will be better if we can use searchResourceFor from DefaultRepository.

We need to create another of this method, but with TokenClientParam.
https://github.com/opensrp/fhircore/blob/47aa32c2281ced24dd858f5647817b6d5c4d88d8/android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt#L85